### ISSUE TITLE
No node features

### ISSUE THREAD

**Issue by bkj** [tkipf/gcn#10]
Hi -- I'm interested in using the GCN in a situation where the node's don't have any features associated with them (besides the label).  It seems like the thing to do is set `featureless=True` in the first layer in `models.py`, and then change the line in `train.py` to 

```
model = model_func(placeholders, input_dim=adj.shape[0], logging=True)
```

This runs, but the performance isn't great.   Any thoughts?  I would expect this to perform somewhere in the neighborhood of `deepwalk`, but I may be misunderstanding the algorithm

Thanks
Ben

**Comment by bkj** [comment]
My bad -- I confused `planetoid` and `deepwalk` when I was reading the paper -- the results w/ `featureless` are actually roughly equivalent to the cited `deepwalk` numbers.  Regardless, I would expect GCN to do _better_ than `deepwalk`, since the labels are available as the node embeddings are learned.  Perhaps the structure of these graphs are modular enough that that doesn't actually matter.

**Comment by tkipf** [comment]
Thanks for your question. Note that by setting `featureless=True`, dropout will be ignored on this layer (as the input to the layer is bypassed altogether). This can significantly hurt generalization performance. You can instead simply pass an identity matrix as input if you want to train without features and use dropout at the same time.

I haven't run experiments without input features for this particular model, so I'm not quite sure what the optimal hyperparameter setting will be in this case and what kind of improvements you can expect over baselines such as DeepWalk.

**Comment by vinhqdang** [comment]
Could you give a sample of trained data? Where do the features come from?

**Comment by vinhqdang** [comment]
Suppose that I have:

- an N by N adjacency matrix (N is the number of nodes),
- an N by D feature matrix (D is the number of features per node), and
- an N by E binary label matrix (E is the number of classes).

in text files. 

How can I feed them into gcn?

**Comment by tkipf** [comment]
The code in its current form expects the adjacency and feature matrix to be a sparse CSR matrix (scipy.sparse.csr_matrix) and the label matrix as a numpy matrix.
